src/test/java/teammates/sqlui/webapi/GetNotificationsActionTest.java

```diff
@@ -0,0 +1,306 @@
+package teammates.sqlui.webapi;
+
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.Const;
+import teammates.storage.sqlentity.Notification;
+import teammates.ui.output.NotificationData;
+import teammates.ui.output.NotificationsData;
+import teammates.ui.webapi.GetNotificationsAction;
+import teammates.ui.webapi.JsonResult;
+
+/**
+ * SUT: {@link GetNotificationsAction}.
+ */
+public class GetNotificationsActionTest extends BaseActionTest<GetNotificationsAction> {
+
+    @Override
+    String getActionUri() {
+        return Const.ResourceURIs.NOTIFICATIONS;
+    }
+
+    @Override
+    String getRequestMethod() {
+        return GET;
+    }
+
+    @BeforeMethod
+    public void baseClassSetup() {
+        loginAsAdmin();
+    }
+
+    @Test
+    protected void testAccessControl_instructorAccessStudentNotification_shouldFail() {
+        InstructorAttributes instructor = typicalBundle.instructors.get("instructor1OfCourse1");
```

Comment:
Same as the other PRs, please use the methods from `src/test/java/teammates/test/BaseTestCase.java`